server/vitals.js

```diff
@@ -140,28 +140,33 @@ async function checkButtonBatteries() {
   }
 }
 
-/* 
-logClientMessage expects an object containing key-values that represent clients with recently disconnected and/or reconnected button(s)
-Each key is a clientID that holds an object with the client information and arrays of any disconnected and/or reconnected buttons
-The function will compile all this information into one message per client, and log the message to Sentry
-*/
-function logClientMessage(clientButtonStatusChanges) {
-  // Loop through each client to create the log message
+// Send button status changes as text messages to every client that has them.
+// This function expects an object (clientButtonStatusChanges) with client IDs as keys, and each each value an object with members:
+// - client: The client object
+// - disconnectedButtons: An array of button display names that refer to buttons that have recently disconnected
+// - reconnectedButtons: An array of button display names that refer to buttons that have recently reconnected
+function sendClientButtonStatusChanges(clientButtonStatusChanges) {
+  // Loop through each client to create the Twilio messages
   Object.values(clientButtonStatusChanges).forEach(({ client, disconnectedButtons, reconnectedButtons }) => {
-    const clientDisplayName = client.displayName
-    let buttonLogMessage = ''
+    let message = [i18next.t('buttonStatusChangeStart', { lng: client.language, clientDisplayName: client.displayName })]
 
     if (disconnectedButtons.length > 0) {
-      const buttonNames = disconnectedButtons.sort().join(', ')
-      buttonLogMessage += ` The following buttons have been disconnected: ${buttonNames}.`
+      const buttonDisplayNames = disconnectedButtons.sort().join(', ') // sorted alphabetically
+      message.push(i18next.t('buttonStatusChangeDisconnected', { lng: client.language, buttonDisplayNames }))
     }
 
     if (reconnectedButtons.length > 0) {
-      const buttonNames = reconnectedButtons.sort().join(', ')
-      buttonLogMessage += ` The following buttons have been reconnected: ${buttonNames}.`
+      const buttonDisplayNames = reconnectedButtons.sort().join(', ') // sorted alphabetically
+      message.push(i18next.t('buttonStatusChangeReconnected', { lng: client.language, buttonDisplayNames }))
     }
 
-    helpers.logSentry(`Button status change for: ${clientDisplayName}.${buttonLogMessage}`)
+    // join the message parts with spaces
+    message = message.join(' ')
+
+    // send SMS text messages to each of the client's heartbeat phone numbers
+    client.heartbeatPhoneNumbers.forEach(phoneNumber => {
+      twilioHelpers.sendTwilioMessage(phoneNumber, client.fromPhoneNumber, message)
+    })
```

Comment:
This might have gotten lost in all the work that Tara did, but these messages should be sent to all the heartbeatPhones and all the ResponderPhones. We want the Responders to know about it because they are the ones on the ground who need to know that a room is currently being unmonitored. This will make it match the list of recipients when a Gateway goes down.

Reply:
Gotcha!